src/Site.js

```diff
@@ -540,13 +539,17 @@ class Site {
       }
 
       /*
-       This is to prevent the first nunjuck call from converting {{baseUrl}} to an empty string
-       and let the baseUrl value be injected later.
+       We retrieve the baseUrl of the (sub)site by appending the relative to the configured base url
+       i.e. We ignore the configured baseUrl of the sub sites.
        */
-      this.variablePreprocessor.addUserDefinedVariable(base, 'baseUrl', '{{baseUrl}}');
+      const siteRelativePathFromRoot = utils.ensurePosix(path.relative(this.rootPath, base));
```

Comment:
This is the main functional change, the rest are just necessary changes due to the tight coupling of `xmlMode`/rendering order/using it to unescape escaped html from removing `resolveBaseUrl`

We simply add the correct baseUrl for each (sub)site as a variable directly into `variablePreprocessor` (previously `userDefinedVariablesMap`)